src/Sky.cc

```diff
@@ -0,0 +1,235 @@
+double Sky::Sunset() const
+{
+  return this->dataPtr->sunset;
+}
+
+/////////////////////////////////////////////////
+void Sky::SetSunset(double _sunset)
+{
+  this->dataPtr->sunset = _sunset;
+}
+
+/////////////////////////////////////////////////
+double Sky::CloudSpeed() const
+{
+  return this->dataPtr->cloudSpeed;
+}
+
+/////////////////////////////////////////////////
+void Sky::SetCloudSpeed(double _speed)
+{
+  this->dataPtr->cloudSpeed = _speed;
+}
+
+/////////////////////////////////////////////////
+ignition::math::Angle Sky::CloudDirection() const
+{
+  return this->dataPtr->cloudDirection;
+}
+
+/////////////////////////////////////////////////
+void Sky::SetCloudDirection(const ignition::math::Angle &_angle)
+{
+  this->dataPtr->cloudDirection = _angle;
+}
+
+/////////////////////////////////////////////////
+double Sky::CloudHumidity() const
+{
+  return this->dataPtr->cloudHumidity;
+}
+
+/////////////////////////////////////////////////
+void Sky::SetCloudHumidity(double _humidity)
+{
+  this->dataPtr->cloudHumidity = _humidity;
+}
+
+/////////////////////////////////////////////////
+double Sky::CloudMeanSize() const
+{
+  return this->dataPtr->cloudMeanSize;
+}
+
+/////////////////////////////////////////////////
+void Sky::SetCloudMeanSize(double _size)
+{
+  this->dataPtr->cloudMeanSize = _size;
+}
+
+/////////////////////////////////////////////////
+ignition::math::Color Sky::CloudAmbient() const
+{
+  return this->dataPtr->cloudAmbient;
+}
+
+/////////////////////////////////////////////////
+void Sky::SetCloudAmbient(const ignition::math::Color &_ambient)
+{
+  this->dataPtr->cloudAmbient = _ambient;
+}
+
+/////////////////////////////////////////////////
+Errors Sky::Load(ElementPtr _sdf)
+{
+  Errors errors;
+
+  this->dataPtr->sdf = _sdf;
+
+  this->dataPtr->time = _sdf->Get<double>("time", this->dataPtr->time).first;
+  this->dataPtr->sunrise =
+      _sdf->Get<double>("sunrise", this->dataPtr->sunrise).first;
+  this->dataPtr->sunset =
+      _sdf->Get<double>("sunset", this->dataPtr->sunset).first;
+
+  if ( _sdf->HasElement("clouds"))
+  {
+    sdf::ElementPtr cloudElem = _sdf->GetElement("clouds");
+    this->dataPtr->cloudSpeed =
+        cloudElem->Get<double>("speed", this->dataPtr->cloudSpeed).first;
+    this->dataPtr->cloudDirection =
+        cloudElem->Get<ignition::math::Angle>("direction",
+        this->dataPtr->cloudDirection).first;
+    this->dataPtr->cloudHumidity =
+        cloudElem->Get<double>("humidity", this->dataPtr->cloudHumidity).first;
+    this->dataPtr->cloudMeanSize =
+        cloudElem->Get<double>("mean_size", this->dataPtr->cloudMeanSize).first;
+    this->dataPtr->cloudAmbient =
+        cloudElem->Get<ignition::math::Color>("ambient",
+        this->dataPtr->cloudAmbient).first;
+  }
+
+  // Check that the provided SDF element is a <sky> element.
+  // This is an error that cannot be recovered, so return an error.
+  if (_sdf->GetName() != "sky")
```

Comment:
Should this be checked before the inner elements are checked?

Reply:
moved check to beginning of function. 05e739a